mrp_product_variants/models/mrp_bom.py

```diff
@@ -86,6 +86,26 @@ def _bom_explode(self, bom, product, factor, properties=None, level=0,
             master_bom=master_bom, production=production)
         return result, result2
 
+    def _check_product_suitable(self, check_attribs, component_attribs):
+        """ Check if component is suitable for given attributes
+        @param production_attribs: Production variant attribute id list
+        @param component_attribs: Component defined attributes to check
+        @return: Component validity
+        """
+        if not check_attribs:
+            return True
+        attrib_group = {}
+        for comp_atribut in component_attribs:
```

Comment:
You can use a handy tool for this:
```
from itertools import groupby

attrib_group = groupby(component_attribs, attribgetter('attribute_id'))
```
You will get a list of recordsets, but you can get ids with a list comprehension:
```
[x.id for x in recordset_list]
```